src/SparseMatrix.chpl

```diff
@@ -0,0 +1,346 @@
+module SparseMatrix {
+
+  use CommDiagnostics, Time;
+  public use SpsMatUtil;
+
+  config const countComms=false,
+               printTimings=false;
+
+  // sparse, outer, matrix-matrix multiplication algorithm; A is assumed
+  // CSC and B CSR
+  //
+  proc sparseMatMatMult(A, B) {
+    if countComms then startCommDiagnostics();
+    var time: stopwatch;
+    time.start();
+
+    var spsData: sparseMatDat;
+
+    sparseMatMatMult(A, B, spsData);
+
+    var C = makeSparseMat(A.domain.parentDom, spsData);
+
+    const elapsed = time.elapsed();
+
+    if countComms {
+      stopCommDiagnostics();
+      printCommDiagnosticsTable();
+      writeln();
+    }
```

Comment:
We should probably remove this debug output for production (along with the timings)